compiler/src/main/kotlin/io/spine/protodata/plugin/View.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.protodata.plugin
+
+import io.spine.base.EntityState
+import io.spine.protobuf.ValidatingBuilder
+import io.spine.server.DefaultRepository
+import io.spine.server.entity.model.EntityClass
+import io.spine.server.projection.Projection
+import io.spine.server.projection.ProjectionRepository
+import io.spine.server.projection.model.ProjectionClass
+import io.spine.server.route.EventRoute
+import io.spine.server.route.EventRouting
+import kotlin.reflect.KClass
+
+/**
+ * A view on the Protobuf sources.
+ *
+ * A view is constructed from the events obtained from the Protobuf Compiler, or generated by
+ * a [Policy].
+ *
+ * Listen to the events using single parameter methods annotated with
+ * [@Subscribe][io.spine.core.Subscribe]. In these methods, change the state of the view via
+ * the `builder()` method.
+ *
+ * We recommend to use `internal` access for the subscriber methods in Kotlin and package-private —
+ * in Java. The methods are not `private`, as they are invoked by the framework, and no `public`,
+ * as they must not be called directly.
+ *
+ * Events from the Protobuf compiler should be marked as [@External][io.spine.core.External], while
+ * events from policies should not.  See the whole list of Protobuf compiler events
+ * in `spine/protodata/events.proto`.
+ *
+ * Example:
+ * ```
+ * class MyView : View<FieldName, Field, Field.Builder> {
+ *
+ *     @Subscribe
+ *     internal fun on(event: SomethingHappened) {
+ *         // Change the view via `builder()`.
```

Comment:
Did you have a chance try the `TransactionalEntity.update()` extension recently introduced in `core-java`? It's declared in the `TransactionalEntityExtensions.kt` under the `server`. A block like this:

```kotlin
@Subscribe
internal fun on(event: SomethingHappened) {
    update {
         // Set props here.
    }
}
```
would look more natural for Kotlin.

Reply:
We're still not at the point when we can release `bootstrap`, as far as I can tell.
Collecting all the right artifacts manually is a non-trivial exercise, which I'd very much like to avoid.
We'll migrate ProtoData to Spine 2.0 and as soon as `bootstrap` is available.